File: hw_1/Agcali_AlpTugrul.lua
workedTime = {["Alp"] = "8 hours", ["Tuna"] = "hours"}
for i,v in pairs(workedTime) do print(i,v) end
print(workedTime["Alp"])
workedTime["Emrehan"] = "8 hours"
for i,v in pairs(workedTime) do print(i,v) end
workedTime["Alp"] = nil
for i,v in pairs(workedTime) do print(i,v) end
workedTime["Tuna"] = "8 hours"
for i,v in pairs(workedTime) do print(i,v) end
print(workedTime["Alp"] ~= nil)
print(workedTime["Tuna"] ~= nil)
function table_contains(tbl, x)
    found = false
    for _, v in pairs(tbl) do
        if v == x then 
            found = true 
        end
    end
    return found
end
print(table_contains(workedTime, "8 hours"))
print(table_contains(workedTime, "hours"))
function foo(key, value)
    print(key, value)
end
for i,v in pairs(workedTime) do foo(i,v) end

File: hw_1/Agcali_AlpTugrul.rs
use std::collections::HashMap;
fn foo(key: &str, val: &str){
    println!("{} {}", key, val);
}
fn main() {
    let mut workedTime=HashMap::new();
    workedTime.insert("Alp", "8 hours");
    workedTime.insert("Tuna", "hours");
    println!("{:?}",workedTime);
    println!("{:?}", workedTime.get(&"Alp"));
    workedTime.insert("Emrehan", "8 hours");
    println!("{:?}",workedTime);
    workedTime.remove(& "Alp");
    println!("{:?}",workedTime);
    workedTime.insert("Tuna","8 hours"); 
    println!("{:?}",workedTime);
    println!("{:?}", workedTime.contains_key( & "Alp"));
    println!("{:?}", workedTime.contains_key( & "Tuna"));
    println!("{:?}", workedTime.values().any(|&val| val == "8 hours"));
    println!("{:?}", workedTime.values().any(|&val| val == "hours"));
    for (key, val) in workedTime.iter() {
        foo(key,val);
    }
}